software/bsg_manycore_lib/bsg_set_tile_x_y.h

```diff
@@ -7,6 +7,7 @@ extern int __bsg_grid_dim_x;	  //The X Dimensions of the grid of tile groups
 extern int __bsg_grid_dim_y;	  //The Y Dimensions of the grid of tile groups
 extern int __bsg_tile_group_id_x; //The X Cord of the tile group within the grid
 extern int __bsg_tile_group_id_y; //The Y Cord of the tile group within the grid
+extern int __bsg_tile_group_id;   //The flat tile group id within the grid
```

Comment:
these definitions seem little confusing.. does the grid mean the entire manycore array?

Reply:
I agree with tommy. Is there a place that `grid` is defined? Perhaps it should be defined in this file.

Reply:
Good point, I should have explained more. The grid is actually a 2D structure or array of tile groups that represent the entire application, with each element being a separate tile group with X,Y coordinates inside the grid. The term came from CUDA, where a grid is a 3D array of thread blocks.
I added some comments on the top of the file, what do you think?

Reply:
Seems good to me. @tommydcjung?